semgrep-core/src/core/ast/Guess_lang.ml

```diff
@@ -0,0 +1,187 @@
+(*
+   Guess whether a given file is indeed written in the specified
+   programming language.
+*)
+
+(*
+   Evaluation will be left-to-right and lazy, using the usual (&&) and (||)
+   operators.
+*)
+type test =
```

Comment:
Do we need this mini-language?

Reply:
Hmm, OK, I get why you added it. You could also have defined some custom combinators to work with `'a -> bool`, e.g.:
```
let (&&$) f g = fun x -> f x && g x 
```

Reply:
yeah, it was one or the other. I like the feel of the `And`, `Or`, etc, probably because they're clearer than punctuation-based operators.